Essentials/src/main/java/com/earth2me/essentials/Jails.java

```diff
@@ -318,5 +353,13 @@ public void onJailPlayerJoin(final PlayerJoinEvent event) {
                 future.completeExceptionally(ex);
             }
         }
+
+        private boolean shouldIgnore(final Player base) {
+            if (base.getUniqueId().version() <= 2) {
+                return true;
+            }
+
+            return !ess.getOnlineModeProvider().isMaybeOfflineMode() && base.getUniqueId().version() == 3;
```

Comment:
Why do we care here about offline mode?

Reply:
if the server is maybe offline mode, we should still run the listener on v3 uuids as it may be real players not uuids